src/Symfony/Component/Form/Util/ParamFilesMerger.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Util;
+
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * @author Priyadi Iman Nurcahyo <priyadi@rekalogika.com>
+ *
+ * @internal
+ */
+class ParamFilesMerger
```

Comment:
Instead of introducing this new internal class the code should be moved into private methods in `FormUtil` as the class is only used there.

Reply:
The class represents a node in the tree. It is used by itself to traverse the entire tree. The task will be very complicated without a dedicated class.

https://github.com/symfony/symfony/pull/54324/files/30c95b8c9396624a98f6e44b0f83889b77112ab7#diff-5983ae1bcb319a7efa00aa62df7a2bcfaeaabd1894959ed7994d30a79f2281deR120

Reply:
I have sent a PR against your fork which inlines the methods from the `ParamFilesMerger` class.

Reply:
I have merged your PR